### PR TITLE
wasm: Add native support for object.get builtin function

### PR DESCRIPTION
**Pull request by ashutosh-narkar**
Signed-off-by: Ashutosh Narkar <anarkar4387@gmail.com>

### CODE REVIEW

File: internal/compiler/wasm/wasm.go
Comment (srenatus) [inline internal/compiler/wasm/wasm.go:114]: [nit] If `object.filter` is `opa_object_filter`, shouldn't we use `opa_object_get` here? 🤔 I take it we already have an `opa_object_get` use for internal purposes; but then maybe we should rename filter to `opa_obj_filter`; or rename the internal function...

File: wasm/tests/test.c
Comment (srenatus) [inline wasm/tests/test.c:1701]: ❔ Worth adding a case for the non-object operand 1? What do you think?

**Review by srenatus — DISMISSED** [review]
LGTM! Two minor things inline, not blocking.

File: internal/compiler/wasm/wasm.go
Comment (tsandall) [reply internal/compiler/wasm/wasm.go:114]: How about `builtin_object_get`, `builtin_object_filter`, etc. This way we can easily tell whether it's a function that implements a built-in. I agree that `opa_obj_get` is suboptimal in terms of consistency...

File: wasm/tests/test.c
Comment (tsandall) [reply wasm/tests/test.c:1701]: +1 from a coverage POV, it would be worth just adding that case.

**Review by tsandall — COMMENTED** [review]
LGTM modulo naming. Also, we don't have to commit the generated code anymore as we have a workflow that will trigger after this gets merged and update it automatically.

**Review by srenatus — APPROVED** [review]
Thanks for addressing my comments! 👍

**Comment by ashutosh-narkar** [comment]
Closing this in favor of https://github.com/open-policy-agent/opa/pull/2995. All comments have been addressed in https://github.com/open-policy-agent/opa/pull/2995.

**Pull request closed without merging**
